src/main/java/DukeException.java

```diff
@@ -0,0 +1,40 @@
+public class DukeException {
```

Comment:
Since DukeException is an exception class, shouldn't it extend the Exception class? This might change a bit of the error handling part of the code since you haven't thrown DukeException, but called the functions from this class for all the error cases... Consider handling the exceptions with try, catch, throw instead of using the class to call the functions and then using `return;` ?
https://nus-cs2113-ay2021s2.github.io/website/schedule/week5/topics.html#c-to-java-exceptions-how-to-use-exceptions